### DISCUSSION TITLE
need help with dll missing error

### DISCUSSION THREAD

**Discussion by vtminhh** [Dadangdut33/Screen-Translate#12] [Q&A] [answered]
i tried to launch the program on a windows 7 PC but ran into this error prompt:

> The program can't start because api-ms-win-core-path-l1-1-0.dll is missing from your computer. Try reinstalling the program to fix this problem.

i've tried system files check + reinstall or download the dll file but couldnt fix the problem

**Comment by vtminhh** [comment]
it seems python 3.9 doesnt work for windows 7. so the only option is to use win 10?

**Reply by Dadangdut33** [reply]
I have done a little bit of some searching and that shows as an answer to me too.
Python 3.9 does not support windows 7 and upgrading to windows 10 seems to be the only way to fix it, **but** you might be able to build the .exe yourself. Have you tried it?

## Here is some step to do it:
1. You can install python 3.5+
2. Clone the repo then do a `pip install -r requirement.txt`.
3. Test the source code, check if the programs could run or not by running the main file `python ScreenTranslate.py`
4. If everything works correctly, you can then do either `pyinstaller ScreenTranslate_Console.spec` or `pyinstaller ScreenTranslate_NoConsole.spec`.
5. The .exe file should be available in dist/ file 


Let me know if it works

**Comment by Dadangdut33** [comment, marked as answer]
Another possible solution might aslo be available at this [StackOverflow discussion](https://stackoverflow.com/questions/66718131/software-based-on-python-3-9-is-not-working-on-windows-7) which leads to [this repository](https://github.com/nalexandru/api-ms-win-core-path-HACK)

**Reply by vtminhh** [reply]
oh yes i found about it just now. i didnt know to code so i couldnt compile the program myself. the dll file from that link worked and STL now runs fine, but there's a small problem, the settings and history windows appear to be not large enough to contain the UI so some text and buttons are cut off:
![2022-04-07-000019](https://user-images.githubusercontent.com/101307929/162200855-ea1409b6-dd4d-4ade-a339-e8e91c1941f4.png)
![2022-04-07-000021](https://user-images.githubusercontent.com/101307929/162200860-68399232-a824-4645-8e54-ad0cfe31f214.png)

**Reply by Dadangdut33** [reply]
That problem seems to happen if your monitor scaling is not set to 100%

For the history window, you can resize it but unfortunately, I made the setting windows unresizeable. I will release an updated version with a resizable setting menu asap.
